network/dime_message.ksy

```diff
@@ -0,0 +1,74 @@
+meta:
+  id: dime_message
+  title: DIME (Direct Internet Message Encapsulation) Message
+  xref:
+    wikidata: Q1227457
+  license: CC0-1.0
+  bit-endian: be
+  endian: be
+doc: |
+  Direct Internet Message Encapsulation (DIME)
+  is an old Microsoft specification for sending and receiving
+  SOAP messages along with additional attachments,
+  like binary files, XML fragments, and even other
+  SOAP messages, using standard transport protocols like HTTP.
+doc-ref: 
+  - http://xml.coverpages.org/draft-nielsen-dime-02.txt
+  - https://docs.microsoft.com/en-us/archive/msdn-magazine/2002/december/sending-files-attachments-and-soap-messages-via-dime
+seq:
+  - id: records
+    type: record
+    repeat: eos
+types:
+  padding:
+    doc: padding to the next 4-byte boundary
+    seq:
+      - id: boundary_padding
+        size: (- _io.pos) % 4
+  record:
+    doc: each individual fragment of the message
+    seq:
+      - id: version
+        type: b5
+      - id: is_first_record
+        type: b1
+      - id: is_last_record
+        type: b1
+      - id: is_chunck_record
+        type: b1
+      - id: type_format
+        enum: type_formats
+        type: b4
+      - id: reserved
+        type: b4
+      - id: len_options
+        type: u2
+      - id: len_id
+        type: u2
+      - id: len_type
+        type: u2
+      - id: len_data
+        type: u4
+      - id: options
+        size: len_options
```

Comment:
Do `options` have some internal structure?

Reply:
Actually they do! Again, from what I read, it was never actually used for anything useful (to my knowledge there aren't any registered types), but is still in the spec and so it should be it struct as well...

I'm adding support for this in the next commit.
I don't know how to test it however, can't find a dime file using these options.

Reply:
I'm crafting a custom file to test this.

Reply:
Alright, it works.

I noticed that even the original `scanner.dump` includes an option field in `records[1]`, but I edited the file to include multiple options (in `records[0]`) and it works as well.
[scanner_withoptions.dump.gz](https://github.com/kaitai-io/kaitai_struct_formats/files/5894723/scanner_withoptions.dump.gz)